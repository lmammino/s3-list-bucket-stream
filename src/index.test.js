const S3ListBucketStream = require(process.env.BUILD ? '../pkg/dist-node/index.js' : './index')

class MockS3 {
  constructor (mockedPages, shouldFail = false) {
    this._mockedPages = mockedPages
    this._currentPage = 0
  }

  listObjectsV2 (params, cb) {

  }
}

test('It should emit all the files from different pages', (done) => {
  expect(true).toBe(true)
  expect(S3ListBucketStream).toBeTruthy()
  expect(MockS3).toBeTruthy()
  done()
})
