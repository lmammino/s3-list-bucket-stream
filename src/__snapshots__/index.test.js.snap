// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should emit all the files from different pages 1`] = `
Array [
  Object {
    "ETag": "etag-0",
    "Key": "0.jpg",
    "LastModified": "2019-02-19T19:35:20.892Z",
    "Size": 0,
    "StorageClass": "STANDARD",
  },
  Object {
    "ETag": "etag-1",
    "Key": "1.jpg",
    "LastModified": "2019-02-19T19:35:20.892Z",
    "Size": 100,
    "StorageClass": "STANDARD",
  },
  Object {
    "ETag": "etag-2",
    "Key": "2.jpg",
    "LastModified": "2019-02-19T19:35:20.892Z",
    "Size": 200,
    "StorageClass": "STANDARD",
  },
  Object {
    "ETag": "etag-3",
    "Key": "3.jpg",
    "LastModified": "2019-02-19T19:35:20.892Z",
    "Size": 300,
    "StorageClass": "STANDARD",
  },
  Object {
    "ETag": "etag-4",
    "Key": "4.jpg",
    "LastModified": "2019-02-19T19:35:20.892Z",
    "Size": 400,
    "StorageClass": "STANDARD",
  },
  Object {
    "ETag": "etag-5",
    "Key": "5.jpg",
    "LastModified": "2019-02-19T19:35:20.892Z",
    "Size": 500,
    "StorageClass": "STANDARD",
  },
]
`;

exports[`The stream should pause if reader buffer is full 1`] = `
Array [
  Object {
    "ETag": "etag-0",
    "Key": "0.jpg",
    "LastModified": "2019-02-19T19:35:20.892Z",
    "Size": 0,
    "StorageClass": "STANDARD",
  },
  Object {
    "ETag": "etag-1",
    "Key": "1.jpg",
    "LastModified": "2019-02-19T19:35:20.892Z",
    "Size": 100,
    "StorageClass": "STANDARD",
  },
  Object {
    "ETag": "etag-2",
    "Key": "2.jpg",
    "LastModified": "2019-02-19T19:35:20.892Z",
    "Size": 200,
    "StorageClass": "STANDARD",
  },
  Object {
    "ETag": "etag-3",
    "Key": "3.jpg",
    "LastModified": "2019-02-19T19:35:20.892Z",
    "Size": 300,
    "StorageClass": "STANDARD",
  },
  Object {
    "ETag": "etag-4",
    "Key": "4.jpg",
    "LastModified": "2019-02-19T19:35:20.892Z",
    "Size": 400,
    "StorageClass": "STANDARD",
  },
  Object {
    "ETag": "etag-5",
    "Key": "5.jpg",
    "LastModified": "2019-02-19T19:35:20.892Z",
    "Size": 500,
    "StorageClass": "STANDARD",
  },
]
`;
